subsys/bluetooth/host/direction.c

```diff
@@ -389,62 +389,90 @@ void hci_df_prepare_connectionless_iq_report(struct net_buf *buf,
 }
 #endif /* CONFIG_BT_DF_CONNECTIONLESS_CTE_RX */
 
-#if defined(CONFIG_BT_CTLR_DF_CONN_CTE_RSP)
-/* @brief Function sets CTE parameters for connection object
+#if defined(CONFIG_BT_DF_CONNECTION_CTE_TX)
+static bool validate_conn_cte_tx_params(const struct bt_df_conn_cte_tx_param *params)
+{
+	if (!(params->cte_types & BT_DF_CTE_TYPE_ALL)) {
+		return false;
+	}
+
+	/* If AoD is not enabled, ant_ids are ignored by controller:
+	 * BT Core spec 5.2 Vol 4, Part E sec. 7.8.84.
+	 */
+	if (params->cte_types & BT_HCI_LE_AOD_CTE_RSP_1US ||
+	    params->cte_types & BT_HCI_LE_AOD_CTE_RSP_2US) {
+		if (params->num_ant_ids < BT_HCI_LE_SWITCH_PATTERN_LEN_MIN ||
```

Comment:
I assume you use two `if` statements instead of `&&` for readability, or was this unintentional?

Reply:
Actually I do not remember ;-). I've removed second if statement and added one more thing: `!BT_FEAT_LE_ANT_SWITCH_TX_AOD(bt_dev.le.features`.

If the AOD feature is not supported by Controller, there is no sense to call this command.